rpc/ethereum/namespaces/eth/api.go

```diff
@@ -673,23 +676,13 @@ func (e *PublicAPI) GetTransactionByBlockHashAndIndex(hash common.Hash, idx hexu
 	}
 
 	i := int(idx)
-	if i >= len(resBlock.Block.Txs) {
+	ethMsgs := e.getEthereumMsgFromTendermintBlock(resBlock)
+	if i >= len(ethMsgs) {
 		e.logger.Debug("block txs index out of bound", "index", i)
 		return nil, nil
 	}
 
-	txBz := resBlock.Block.Txs[i]
-	tx, err := e.clientCtx.TxConfig.TxDecoder()(txBz)
-	if err != nil {
-		e.logger.Debug("decoding failed", "error", err.Error())
-		return nil, fmt.Errorf("failed to decode tx: %w", err)
-	}
-
-	msg, err := evmtypes.UnwrapEthereumMsg(&tx)
-	if err != nil {
-		e.logger.Debug("invalid tx", "error", err.Error())
-		return nil, err
-	}
+	msg := ethMsgs[i]
```

Comment:
are we assuming that the index is always the index from set of Ethereum txs instead of the index of the tx in the block? cc: @tomtau

Reply:
Yeah since `eth_getBlockByXX` hide all non ethereum tx, I think we can use this assumption

ref https://github.com/tharsis/ethermint/blob/main/rpc/ethereum/backend/backend.go#L352

Reply:
to give more background to this issue we found that there was missmatch value between 
`eth_getBlockTransactionCountByNumber` and the result in `eth_getBlockByNumber`

Reply:
can we put that as a code comment somewhere? @crypto-facs

Reply:
Done 👌